### PR TITLE
[fix](mtmv)fix generate partition name illegality when partition value contains colon

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java [hunk 1]
```diff
@@ -287,8 +289,9 @@ public static boolean isSyncWithPartition(MTMV mtmv, Long mtmvPartitionId,
      */
     public static String generatePartitionName(PartitionKeyDesc desc) {
         String partitionName = "p_";
-        partitionName += desc.toSql().trim().replaceAll("\\(|\\)|\\-|\\[|\\]|'|\\s+", "")
-                .replaceAll("\\(|\\)|\\,|\\[|\\]", "_");
+        Pattern pattern = Pattern.compile("[^a-zA-Z0-9,]");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java:292]: make this variable static and final?

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java [hunk 2]
```diff
@@ -287,8 +289,9 @@ public static boolean isSyncWithPartition(MTMV mtmv, Long mtmvPartitionId,
...
+        Matcher matcher = pattern.matcher(desc.toSql());
+        partitionName += matcher.replaceAll("").replaceAll("\\,", "_");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java:294]: may generate same name for different partition

Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java:294]: Temporarily put on hold

**Review by morrySnow — APPROVED** [review]

**Review by Jibing-Li — APPROVED** [review]

**Pull request merged**